### ISSUE TITLE
prefixItems is unsupported (OpenAPI 3.1)

### ISSUE THREAD

**Issue by syniex** [anymaniax/orval#890]
## What are the steps to reproduce this issue?

1. use prefixItems in the openapi (version 3.1)

## What happens?
Orval will crash saying `ovt - Error: All arrays must have an items key define`

## What were you expecting to happen?
Client will be generated

## Any logs, error output, etc?
**Swagger Example** 
![image](https://github.com/anymaniax/orval/assets/90573138/7fa01f19-33cd-47f4-be97-46eac3bbca4e)

## Any other comments?

```json
{
    "openapi": "3.1.0",
    "info": {
        "title": "Fail Example",
        "version": "0.1.0"
    },
    "paths": {
        "/api/v1/building/": {
            "get": {
                "tags": [
                    "Version 1",
                    "Building"
                ],
                "summary": "Get Buildings",
                "operationId": "get_buildings",
                "responses": {
                    "200": {
                        "description": "Successful Response",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/Polygon"
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    "components": {
        "schemas": {
            "Polygon": {
                "properties": {
                    "type": {
                        "const": "Polygon",
                        "title": "Type",
                        "default": "Polygon"
                    },
                    "coordinates": {
                        "items": {
                            "items": {
                                "prefixItems": [
                                    {
                                        "type": "number"
                                    },
                                    {
                                        "type": "number"
                                    }
                                ],
                                "type": "array",
                                "maxItems": 2,
                                "minItems": 2
                            },
                            "type": "array"
                        },
                        "type": "array",
                        "title": "Coordinates"
                    }
                },
                "type": "object",
                "required": [
                    "coordinates"
                ],
                "title": "Polygon",
                "description": "Polygon Model."
            }
        }
    }
}
```

**Comment by dan-garden** [comment]
Having the same issue, everywhere else my spec is valid. Can't find any info online for this specific issue either.

**Comment by GuillaumeQuenneville** [comment]
I can confirm this issue. Using python and pydantic, defining a type with a tuple_positional_schema generates a prefixItems key. Changing the type's json schema to a List instead of a Tuple in my openapi generator works as a temporary (less robust ) fix.

**Comment by rauldeheer** [comment]
Any news on this?

**Comment by Pierstoval** [comment]
Just encountered the same error, as well as countless `must have required property '$ref'` or `must match exactly one schema in oneOf` errors alongside the `Error: All arrays must have an `items` key define` error 😕 

Is there any workaround so far?

**Comment by rauldeheer** [comment]
@Pierstoval I was able to fix it. My Swagger doc is auto-generated by NestJS Swagger. The problem was that arrays of a 'simple type' such as a number or a string didn't have a type in the Swagger JSON. So at first I had:

  ```js
@ApiProperty({ required: false, isArray: true })
@Expose()
numberArray?: number[];
```

Once I changed all those arrays to the following:

```js
@ApiProperty({ required: false, isArray: true, type: Number })
@Expose()
numberArray?: number[];
```

The schema added the correct type and it was instantly fixed. So all I had to do was add the correct `type` property to the `ApiProperty` decorator. Hope this helps!
